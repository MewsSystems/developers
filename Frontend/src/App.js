import React, { useEffect } from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';
import store from './store';
import Home from './pages/Home';
import Detail from './pages/Detail';
import NavBar from './components/layout/NavBar';
import About from './pages/About';
import NotFound from './pages/NotFound';

import 'materialize-css/dist/css/materialize.min.css';
import M from 'materialize-css/dist/js/materialize.min';
import './App.css';


const App = () => {

  useEffect(() => {
    M.AutoInit();
  });

  return (
    <Provider store={store}>
      <Router>
        <NavBar />
        <Switch>
          <Route exact path='/' component={Home} />
          <Route exact path='/about' component={About} />
          <Route exact path='/movie/:id' component={Detail} />
          <Route component={NotFound} />
        </Switch>
      </Router>
    </Provider>
  );
};

export default App;
