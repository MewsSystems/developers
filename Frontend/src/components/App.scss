html,
body {
  margin: 0;
  padding: 0;
}

html {
  background: #000;
  color: #fff;
}

// react-paginate library doesn't provide api to replace its componets
// so I didn't find a way to use styled components with it
.Search-Page-Pagination {
  display: flex;
  list-style: none;
  justify-content: center;
  margin: 2rem 0;

  .previous,
  .next {
    display: none;
  }

  li {
    border: 1px solid #fff;
    margin: 0 0.25rem;
    padding: 0;
    cursor: pointer;

    &.break {
      border: none;
    }

    &.selected {
      background: #fff;
      color: #000;
      border-color: #000;
    }
  }

  a {
    display: block;
    padding: 0.5rem;
  }
}
