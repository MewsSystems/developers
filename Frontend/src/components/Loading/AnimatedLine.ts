import styled from 'styled-components'

export interface AnimatedLineProps {
  width: string
  height: string
}

export const AnimatedLine = styled.div<AnimatedLineProps>`
  width: ${({ width }) => width};
  height: ${({ height }) => height};
  background: linear-gradient(
    90deg,
    rgba(195, 195, 195, 1) 0%,
    rgba(246, 246, 246, 1) 100%
  );
  animation: AnimationName 5s ease infinite;

  @keyframes AnimationName {
    0% {
      background-position: 0% 50%;
    }
    50% {
      background-position: 100% 50%;
    }
    100% {
      background-position: 0% 50%;
    }
  }
`
