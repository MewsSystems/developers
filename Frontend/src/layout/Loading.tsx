import * as React from 'react'
import { keyframes } from '@emotion/core'
import { transparentize } from 'polished'
import styled from '../utils/styled'

export const LoadingSpinner: React.SFC = () => (
  <Spinner>
    <div className="cube1" />
    <div className="cube2" />
  </Spinner>
)

const CubeMove = keyframes`
  25% {
    transform: translateX(42px) rotate(-90deg) scale(0.5);
    -webkit-transform: translateX(42px) rotate(-90deg) scale(0.5);
  } 50% {
    transform: translateX(42px) translateY(42px) rotate(-179deg);
    -webkit-transform: translateX(42px) translateY(42px) rotate(-179deg);
  } 50.1% {
    transform: translateX(42px) translateY(42px) rotate(-180deg);
    -webkit-transform: translateX(42px) translateY(42px) rotate(-180deg);
  } 75% {
    transform: translateX(0px) translateY(42px) rotate(-270deg) scale(0.5);
    -webkit-transform: translateX(0px) translateY(42px) rotate(-270deg) scale(0.5);
  } 100% {
    transform: rotate(-360deg);
    -webkit-transform: rotate(-360deg);
  }
`

const Spinner = styled('div')`
  margin: 100px auto;
  width: 40px;
  height: 40px;
  position: relative;

  & .cube1,
  & .cube2 {
    background-color: ${props => props.theme.colors.brand};
    width: 15px;
    height: 15px;
    position: absolute;
    top: 0;
    left: 0;

    -webkit-animation: ${CubeMove} 1.8s infinite ease-in-out;
    animation: ${CubeMove} 1.8s infinite ease-in-out;
  }

  & .cube2 {
    -webkit-animation-delay: -0.9s;
    animation-delay: -0.9s;
  }
`

export const LoadingOverlay = styled('div')`
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  z-index: 10;
  background: ${props => transparentize(0.25, props.theme.colors.background)};
`

export const LoadingOverlayInner = styled('div')`
  display: flex;
  align-items: center;
  justify-content: center;
  min-height: 200px;
`

type LoadingProps = {
  loading: boolean
}

export const Loading = ({ loading }: LoadingProps) => {
  if (!loading) return null

  return (
    <LoadingOverlay>
      <LoadingOverlayInner>
        <LoadingSpinner />
      </LoadingOverlayInner>
    </LoadingOverlay>
  )
}
