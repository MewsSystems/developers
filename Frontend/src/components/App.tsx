import React from 'react';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import Search from './Search/Search';
import Movie from './Movie/Movie';
import { Provider } from 'react-redux';
import createStore from '../store';
import './App.scss';

const store = createStore();

const App = () => {
  return (
    <Provider store={store}>
      <Router>
        <Switch>
          <Route path="/" exact component={Search} />
          <Route path="/movie/:id" render={({ match }) => <Movie id={match.params.id} />} />
        </Switch>
      </Router>
    </Provider>
  );
};

export default App;
