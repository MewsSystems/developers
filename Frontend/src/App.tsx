import { Layout } from 'components/Layout'
import React, { useEffect } from 'react'
import { Helmet } from 'react-helmet'
import { useTranslation } from 'react-i18next'
import { useThunkDispatch } from 'hooks/useThunkDispatch'
import { hydrateConfiguration } from 'state/actions/configuration'
import { Router } from 'components/Router'
import './i18n'

const App: React.FC = () => {
  const { t } = useTranslation()
  const dispatch = useThunkDispatch()

  useEffect(() => {
    dispatch(hydrateConfiguration())
  }, [dispatch])

  return (
    <>
      <Helmet>
        <title>{t('html-title')}</title>
      </Helmet>
      <Layout>
        <Router />
      </Layout>
    </>
  )
}

export default App
