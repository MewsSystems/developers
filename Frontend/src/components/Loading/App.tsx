import React from 'react'
import styled from 'styled-components'
import { COLORS } from 'constants/colors'
import { Content } from 'components/Content'
import { AnimatedLine } from './AnimatedLine'
import { Spacing } from 'components/Spacing'

const Header = styled.div`
  height: 10.6rem;
  background: ${COLORS.GRAY};
`

export const App: React.FC = () => (
  <>
    <Header />
    <Content>
      <Spacing outer="2rem 0">
        <AnimatedLine width="20rem" height="2rem" />
      </Spacing>
    </Content>
  </>
)
