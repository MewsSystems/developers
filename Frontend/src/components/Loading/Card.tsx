import React from 'react'
import { AnimatedLine } from './AnimatedLine'
import { StyledCard } from 'components/Card'
import { Spacing } from 'components/Spacing'

export const Card: React.FC = () => (
  <StyledCard>
    <Spacing outer="0 0 1rem">
      <AnimatedLine width="100%" height="2rem" />
    </Spacing>
    <AnimatedLine width="100%" height="1rem" />
  </StyledCard>
)
