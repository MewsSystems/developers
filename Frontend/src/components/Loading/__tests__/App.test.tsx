import { App } from 'components/Loading/App'
import { shallow } from 'enzyme'
import React from 'react'

describe('Test Loading App', () => {
  it('Tests snapshot', () => {
    expect(shallow(<App />)).toMatchSnapshot()
  })
})
