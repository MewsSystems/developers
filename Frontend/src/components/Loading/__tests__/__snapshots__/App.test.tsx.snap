// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Loading App Tests snapshot 1`] = `
<Fragment>
  <styled.div />
  <styled.div>
    <Spacing
      outer="2rem 0"
    >
      <styled.div
        height="2rem"
        width="20rem"
      />
    </Spacing>
  </styled.div>
</Fragment>
`;
